pkg/signup/service/signup_service_test.go

```diff
@@ -1257,12 +1273,22 @@ func (s *TestSignupServiceSuite) TestGetSignupByUsernameOK() {
 	require.NoError(s.T(), err)
 
 	// when
-	response, err := s.Application.SignupService().GetSignup(c, "foo", us.Spec.IdentityClaims.PreferredUsername)
+	response, err := svc.GetSignup(c, "foo", us.Spec.IdentityClaims.PreferredUsername)
 
 	// then
 	require.NoError(s.T(), err)
 	require.NotNil(s.T(), response)
 
+	// Confirm the StartDate is the same as the provisionedTime
+	require.Equal(s.T(), provisionedTime.Format(time.RFC3339), response.StartDate)
+
+	// Confirm the end date is about 1 day ago
+	responseEndDate, err := time.Parse(time.RFC3339, response.EndDate)
+	require.NoError(s.T(), err)
+	expectedEndDate := time.Now().Add(-time.Hour * 24)
+	require.WithinDuration(s.T(), expectedEndDate, responseEndDate, time.Minute*5,
+		"endDate in response [%s] not in expected range [%s]", responseEndDate, expectedEndDate.Format(time.RFC3339))
```

Comment:
why do we need to verify that the time is "about" 1 day ago? We do set the `ScheduledDeactivationTimestamp` to a specific time, and the value is copied from the UserSignup status, so we can use `require.Equal` expecting the same timestamp to be set also in the Signup response.
```suggestion
	// Confirm the end date is the same as ScheduledDeactivationTimestamp
	require.Equal(s.T(), us.Status.ScheduledDeactivationTimestamp.Format(time.RFC3339), response.EndDate)
```

Reply:
Updated to check exact timestamp.